<h2>TFSA-2022-135: <code>CHECK</code> fail in <code>DrawBoundingBoxes</code></h2>
<h3>CVE Number</h3>
<p>CVE-2022-36001</p>
<h3>Impact</h3>
<p>When <code>DrawBoundingBoxes</code> receives an input <code>boxes</code> that is not of dtype <code>float</code>, it gives a <code>CHECK</code> fail that can trigger a denial of service attack.
<code>python
import tensorflow as tf
import numpy as np
arg_0=tf.constant(value=np.random.random(size=(1, 3, 2, 3)), shape=(1, 3, 2, 3), dtype=tf.half)
arg_1=tf.constant(value=np.random.random(size=(1, 2, 4)), shape=(1, 2, 4), dtype=tf.float32)
arg_2=''
tf.raw_ops.DrawBoundingBoxes(images=arg_0, boxes=arg_1, name=arg_2)</code></p>
<h3>Patches</h3>
<p>We have patched the issue in GitHub commit <a href="https://github.com/tensorflow/tensorflow/commit/da0d65cdc1270038e72157ba35bf74b85d9bda11">da0d65cdc1270038e72157ba35bf74b85d9bda11</a>.</p>
<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>
<h3>For more information</h3>
<p>Please consult <a href="https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>
<h3>Attribution</h3>
<p>This vulnerability has been reported by 刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology.</p>